cdap-app-fabric/src/test/java/co/cask/cdap/security/SecureStoreTest.java

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.security;
+
+import co.cask.cdap.internal.app.services.http.AppFabricTestBase;
+import co.cask.cdap.internal.guava.reflect.TypeToken;
+import co.cask.cdap.proto.security.SecureStoreCreateRequest;
+import co.cask.cdap.security.store.FileSecureStore;
+import com.google.gson.Gson;
+import org.apache.http.HttpResponse;
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.lang.reflect.Type;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class SecureStoreTest extends AppFabricTestBase {
+  private static final Gson GSON = new Gson();
+  private static Type listType = new TypeToken<ArrayList<FileSecureStore.SecureStoreListEntry>>() { }.getType();
+  private static final String KEY = "Key1";
+  private static final String DESCRIPTION = "This is Key1";
+  private static final String DATA = "Secret1";
+  private static final Map<String, String> PROPERTIES = new HashMap<>();
+  static {
+    PROPERTIES.put("Prop1", "Val1");
+    PROPERTIES.put("Prop2", "Val2");
+  }
+
+  private static final String KEY2 = "Key2";
+  private static final String DESCRIPTION2 = "This is Key2";
+  private static final String DATA2 = "Secret2";
+  private static final Map<String, String> PROPERTIES2 = new HashMap<>();
+  static {
+    PROPERTIES2.put("Prop1", "Val1");
+    PROPERTIES2.put("Prop2", "Val2");
+  }
+
+  @Test
+  public void testCreate() throws Exception {
+    SecureStoreCreateRequest secureStoreCreateRequest = new SecureStoreCreateRequest(KEY, DESCRIPTION, DATA,
```

Comment:
while its nice to split up tests into their own methods, just make sure that we don't have a lot of overhead for setup/teardown.